File: mobile/lib/models/pest_detection.dart
import 'detection_result.dart';

/// Structured pest-detection response (`PestDetectionOut`).
///
/// Status values mirror the backend provider layer:
/// - `MODEL_NOT_CONFIGURED` — no model is bundled; never confused with
///   "no pest" or a real identification.
class PestDetectionResult implements DetectionResultModel {
  const PestDetectionResult({
    required this.success,
    required this.status,
    required this.crop,
    required this.pestName,
    required this.confidence,
    required this.model,
    required this.message,
  });

  static const String statusModelNotConfigured = 'MODEL_NOT_CONFIGURED';

  final bool success;
  final String status;
  final String? crop;
  final String? pestName;
  @override
  final double? confidence;
  final String? model;
  @override
  final String? message;

  @override
  bool get isModelNotConfigured => status == statusModelNotConfigured;

  factory PestDetectionResult.fromJson(Object? json) {
    final map = json as Map<String, dynamic>;
    return PestDetectionResult(
      success: map['success'] as bool? ?? true,
      status: map['status'] as String? ?? statusModelNotConfigured,
      crop: map['crop'] as String?,
      pestName: map['pest_name'] as String?,
      confidence: (map['confidence'] as num?)?.toDouble(),
      model: map['model'] as String?,
      message: map['message'] as String?,
    );
  }
}

File: config/core/schemas/pest_detection.py
"""
KisanAI OS
Pest Detection Schemas
Version: 1.0.0

Future-compatible prediction result. A real model will populate
``pest_name``, ``confidence`` and ``model``; until then these stay
``None`` with ``status = MODEL_NOT_CONFIGURED`` so "no model" is never
confused with "no pest" or "pest detected". No local filesystem paths
are ever exposed.
"""

from typing import Optional

from pydantic import BaseModel


class PestDetectionOut(BaseModel):
    """Structured pest-detection response."""

    success: bool = True
    status: str
    crop: Optional[str] = None
    pest_name: Optional[str] = None
    confidence: Optional[float] = None
    model: Optional[str] = None
    message: Optional[str] = ""
